lib/fastlane/plugin/fueled/actions/define_versions_ios.rb

```diff
@@ -10,7 +10,7 @@ module SharedValues
     class DefineVersionsIosAction < Action
       def self.run(params)
         # Build Number
-        Actions.lane_context[SharedValues::FUELED_BUILD_NUMBER] = Helper::FueledHelper.new_build_number(filter:params[:build_type])
+        Actions.lane_context[SharedValues::FUELED_BUILD_NUMBER] = Helper::FueledHelper.new_build_number(filter: '')
```

Comment:
@notbenoit any particular reason for removing the build_type? The filtering allowed us to maintain separate version numbers for release and snapshot paths

Reply:
Yes, build numbers should be incremented for each build, regardless of the flavor I suppose. Do you have iOS projects relying on this behavior ?

Reply:
It's the equivalent of the BUILD_NUMBER env var managed by CI providers. Here we just use something which is platform agnostic, by relying on the repo itself.

Reply:
@notbenoit Yes, For KMP projects we like to keep this uniform across both platforms. There is a single workflow that builds it for both platforms and I'm not sure about using version numbers that have gaps across the individual build types.
Having build_type agnostic version numbers might be a better fit when we are piping each of the builds into the playstore/appstore. However, we have been using build_types as lanes for internal or client-facing circulation (alpha snapshot, alpha release). 
I feel like it would make more sense if these channels have local sequenced build numbers rather than a global sequence distributed across multiple channels.

Reply:
The clients would see a consistent build number in `alpha release`
0.2.1 (21) 0.2.2(22)
instead of 
0.2.1 (21) 0.2.2 (26).
What do you think?

Reply:
My concern is that you can end up with duplicated build numbers then, for two versions that have nothing in common. Eg 
`0.2.2-Snapshot#12`
and 
`1.0.0-Production#12`
We most often reference build by their build number. So what if I say the QA team to QA build `#12` ? 🤔

Reply:
>I feel like it would make more sense if these channels have local sequenced build numbers rather than a global sequence distributed across multiple channels.

Channels already have dedicated short version numbers by the way. Do we really need to have another channel specific number (the build number) ?

Reply:
Feel free to open a new PR to revert the change if you feel it's not going in the right direction. We'll adapt.

Reply:
Thanks @notbenoit, I'll try to go through our versioning once I'm a bit more time on my hand and see if we can consolidate it better across different kinds of projects(KMP, Flutter and purely native builds) for now what we have works well.